kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderAmendmentDocument.java

```diff
@@ -0,0 +1,49 @@
+package edu.arizona.kfs.module.purap.document;
+
+import org.kuali.kfs.sys.businessobject.Building;
+import edu.arizona.kfs.sys.businessobject.BuildingExtension;
+
+public class PurchaseOrderAmendmentDocument extends org.kuali.kfs.module.purap.document.PurchaseOrderAmendmentDocument {
+	
+    protected String routeCode;
+    protected Building buildingObj;
+
+
+    /**
+     * Gets the routeCodeObj attribute. 
+     * @return Returns the routeCodeObj.
+     */
+    public Building getBuildingObj() {
+        return buildingObj;
+    }
```

Comment:
Method getBuildingObj() should be more robust. Refer back to how you coded method BuildingExtension.getRouteCodeObj().